src/components/overlay/overlay.scss

```diff
@@ -0,0 +1,193 @@
+@use '../core/styles' as sbb;
+
+// Default component properties, defined for :host. Properties which can not
+// travel the shadow boundary are defined through this mixin
+@include sbb.host-component-properties;
+
+:host {
+  @include sbb.scrollbar-variables;
+
+  --sbb-overlay-padding-inline: var(--sbb-spacing-fixed-5x);
+  --sbb-overlay-color: var(--sbb-color-black);
+  --sbb-overlay-background-color: var(--sbb-color-milk);
+  --sbb-overlay-width: 100%;
+  --sbb-overlay-height: 100%;
+  --sbb-overlay-inset: 0 auto auto 0;
+  --sbb-overlay-padding-block: var(--sbb-spacing-responsive-s);
+  --sbb-overlay-animation-duration: var(--sbb-animation-duration-6x);
+  --sbb-overlay-animation-easing: ease;
+  --sbb-overlay-pointer-events: none;
+  --sbb-overlay-content-transition: transform var(--sbb-overlay-animation-duration)
+    var(--sbb-overlay-animation-easing);
+
+  position: fixed;
+  inset: var(--sbb-overlay-inset);
+  z-index: var(--sbb-overlay-z-index, var(--sbb-overlay-z-index));
```

Comment:
little bit funny definition here ;) but we have a naming collision between global sbb-overlay-z-index variable and the z index of this component. renaming the global one would be breaking, renaming this one would be inconsistent regarding the naming. @kyubisation

Reply:
I discussed this with Lukas and we decided to rename the global var to `--sbb-overlay-default-z-index` and document the breaking change.